apps/prairielearn/src/tests/helperExam.ts

```diff
@@ -31,32 +31,30 @@ export const questions = _.keyBy(questionsArray, 'qid');
 // must be the sum of maxPoints in questionsArray, but we hard-code it for reference
 export const assessmentMaxPoints = 94;
 
-export interface ExamLocals {
-  siteUrl: string;
-  baseUrl: string;
-  courseInstanceBaseUrl: string;
-  instructorBaseUrl: string;
-  instructorAssessmentsUrl: string;
-  instructorGradebookUrl: string;
-  questionBaseUrl: string;
-  assessmentsUrl: string;
-  assessmentUrl: string;
-  assessmentInstanceUrl: string;
+export function startExam(locals: {
+  siteUrl?: string;
+  baseUrl?: string;
+  courseInstanceBaseUrl?: string;
+  instructorBaseUrl?: string;
+  instructorAssessmentsUrl?: string;
+  instructorGradebookUrl?: string;
+  questionBaseUrl?: string;
+  assessmentsUrl?: string;
+  isStudentPage?: boolean;
+  totalPoints?: number;
+  assessmentUrl?: string;
+  assessmentInstanceUrl?: string;
 
-  isStudentPage: boolean;
-  totalPoints: number;
-  assessment_id: string;
-  $: cheerio.CheerioAPI;
-  __csrf_token: string;
-  preStartTime: number;
-  postStartTime: number;
-  assessment_instance: {
+  assessment_id?: string;
+  $?: cheerio.CheerioAPI;
+  __csrf_token?: string;
+  preStartTime?: number;
+  postStartTime?: number;
+  assessment_instance?: {
     assessment_id: string;
   };
-  instance_questions: { id: string; qid: string }[];
-}
-
-export function startExam(locals: ExamLocals) {
+  instance_questions?: { id: string; qid: string }[];
```

Comment:
@nwalters512 this is a weird scenario that I'm not used to in TS. The locals object is updated by the function, so these are in essence out parameters (though within the tests, not in the function itself). I set all fields as optional, is this an appropriate way to declare this?